walkthrough/src/main/java/com/robotsandpencils/walkthrough/presentation/main/WalkThroughActivity.java

```diff
@@ -118,6 +117,18 @@ protected void onDestroy() {
     }
 
     @Override
+    public void onResume() {
+        super.onResume();
+        mWalkThroughManager.setSpinner(this);
+    }
+
+    @Override
+    public void onPause() {
+        super.onPause();
+        mWalkThroughManager.setSpinner(null);
```

Comment:
I guess I should have mentioned the only real way to reproduce this is to run the code through MyO2:
- bring up the offer screen by running the app 5 times
- decline an offer
- close the offers screen
- re-enter the pin (the pin screen will be showing after the offers screen if "Don't keep activities" is on)

The spinner wasn't being reset before the new WalkThroughActivity appeared and so the old spinner was being used.